CollectNewspaperKarel.java

```diff
@@ -12,6 +13,84 @@
 
 public class CollectNewspaperKarel extends SuperKarel {
 
-	// You fill in this part
+	public void run() {
+		moveToNewsPaper();
+		pickUpNewsPaper();
+		returnToStartPosition();
+	}
+
+	private void moveToNewsPaper() {
+		// move while front is clear
+		moveToWall();
+		faceSouth();
+		move();
+		faceEast();
+		move();
+	}
+
+	private void pickUpNewsPaper() {
+		pickBeeper();
+	}
+
+	private void returnToStartPosition() {
+		turnMultipleLeft(2);
+		moveToWall();
+		turnMultipleLeft(3);
+		moveToWall();
+		turnMultipleLeft(3);
+	}
+
+	private void moveToWall() {
+		while (frontIsClear()) {
+			move();
+		}
+	}
+
+	private void faceSouth() {
```

Comment:
yea... just noticed that the api had that the notfacing* method... an over sight.